### PR TITLE
Extract dustapp layout

### PR DESCRIPTION
**Pull request by tdraier**
## Description

Hoists `DustAppPageLayout` into a shared layout so that the tab bar persists when navigating between app sub-pages (specification, datasets, runs, settings) — eliminating the flicker caused by the layout unmounting/remounting on each navigation.

### What changed

- **`DustAppPageLayout` is now self-contained**: it fetches the app via `useApp()` and derives the current tab from the route path, instead of receiving `app`/`currentTab` as props.
- **Next.js**: added `dustAppGetLayout()` in `appGetLayout.tsx` (same pattern as `spaceGetLayout`, `adminGetLayout`). All 8 app page files now use it.
- **SPA**: added `DustAppRouterLayout` and nested all app routes under it in `App.tsx`.
- **Simplified 8 page components**: removed `<DustAppPageLayout>` wrappers from 6 pages, removed manual tabs + `useAppLayoutConfig` from `AppSpecificationPage` and `DatasetPage`.

## Tests

- `npx tsgo --noEmit` passes
- ESLint on changed files passes
- Manual: navigate between app tabs — tabs persist without flicker

## Risk

Low. This is a layout restructuring with no business logic changes. The tab rendering and `useAppLayoutConfig` logic moved from individual pages into the shared layout — same behavior, just hoisted.

## Deploy Plan

Standard deploy — no migrations, no feature flags, no dependencies.

### CODE REVIEW

File: front-spa/src/lib/platform.tsx [hunk 1]
```diff
@@ -273,20 +273,25 @@ export function useAppRouter(): AppRouter {
...
+      pathname,
+      asPath: pathname + search + hash,
+      query,
+      isReady,
+      events: routerEvents,
+      // beforePopState is a noop in SPA mode (not supported in React Router)
+      // TODO: Check usage in AppContentLayout and remove it.
+      beforePopState,
+    }),
+    [push, replace, back, reload, pathname, search, hash, query, beforePopState]
```
Comment (PopDaph) [inline front-spa/src/lib/platform.tsx:293]: Can you explain what's the goal of this change? 🙏🏻

**Review by PopDaph — APPROVED** [review]
Code LGTM, let me know if you need help testing something related on front-edge 👍🏻

File: front-spa/src/lib/platform.tsx (hunk 1 above)
Comment (tdraier) [reply front-spa/src/lib/platform.tsx:293]: yes , router was not stable (should be - useRouter should always return the same object) , it was creating a render loop
